### PR TITLE
Prevent malformed URL like /fr/index.php?

### PR DESCRIPTION
**Pull request by SebBareyre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This modification prevents the language from being added for URLs that start with `/index.php?`
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | Visit the page "Order history" and clic on "Details" on one of your order. Finally, check the URL

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9367)

### CODE REVIEW

**Review by SebBareyre — COMMENTED** [review]
add missing space

File: classes/Link.php [hunk 1]
```diff
@@ -960,7 +960,11 @@ public function getPageLink($controller, $ssl = null, $idLang = null, $request =
 
         $uriPath = Dispatcher::getInstance()->createUrl($controller, $idLang, $request, false, '', $idShop);
 
-        return $this->getBaseLink($idShop, $ssl, $relativeProtocol).$this->getLangLink($idLang, null, $idShop).ltrim($uriPath, '/');
+        if (substr($uriPath, 0, 10) === 'index.php?') {
+            return $this->getBaseLink($idShop, $ssl, $relativeProtocol).ltrim($uriPath, '/');
+        } else {
```
Comment (PierreRambaud) [inline classes/Link.php]: else is useless because of returns statements :)

**Review by SebBareyre — COMMENTED** [review]
remove else

**Comment by SebBareyre** [comment]
@PierreRambaud : Any news about this PR ?

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Sorry for the delay :)

**Comment by SebBareyre** [comment]
No problem @PierreRambaud :)

**Comment by ntiepresta** [comment]
Hi, @SebBareyre 

Could you please provide us with more info? We need more details to understand how we can reproduce your issue.
- screenshots for example

Best regards, Reddy

**Comment by SebBareyre** [comment]
[Here](https://pasteboard.co/HwR6ydt.jpg) you'll find a screenshot that shows how you will reproduce the bug. This bug occurs each time a URL called is not rewritten on nginx...

**Comment by ntiepresta** [comment]
Thank you @SebBareyre

**Comment by PierreRambaud** [comment]
Hum look like E2E tests are broken :/

**Comment by SebBareyre** [comment]
@PierreRambaud how can I re-run E2E tests please ?

**Comment by PierreRambaud** [comment]
@SebBareyre If you wait for #9301 it will be easier to execute E2E tests :)

**Comment by mickaelandrieu** [comment]
Looking at stack trace, I've found this error in logs:

[01-Aug-2018 15:02:45 Europe/Paris] PHP Warning:  Base directory /home/travis/build/PrestaShop/PrestaShop/var/cache/prod/purifier does not exist,
                    please create or change using %Cache.SerializerPath in /home/travis/build/PrestaShop/PrestaShop/tools/htmlpurifier/HTMLPurifier.standalone.php on line 15280


This may not be related at all.

Also @PierreRambaud, how can we access the screenshots on travis? I've looked at all builds and I can't manage to get the base64 strings of captures.

Still investigating on this, wait and chill fellows :sunglasses:

**Comment by PierreRambaud** [comment]
@mickaelandrieu You can add a parameter `PRESTASTHOP_SCREENHOST` while building.

**Comment by mickaelandrieu** [comment]
> You can add a parameter PRESTASTHOP_SCREENHOST while building

I don't build anything as the docker image to run e2e tests is broken :(

**Comment by mickaelandrieu** [comment]
Hello @PierreRambaud, would you like to take a closer look at it if you are now able to launch e2e tests using docker.

I still can't :/

**Comment by PierreRambaud** [comment]
@mickaelandrieu You have now the capability to reproduce in local with latest commits.

**Comment by mickaelandrieu** [comment]
@PierreRambaud I guess the docker image forgot some scenarios, isn't it?

>  99 passing (40s) on Docker

> 121 passing (80.50s) on Travis
   1 failing

**Comment by PierreRambaud** [comment]
It runs the documented command `npm run test`. But you can run the test you want to execute inside the docker container.

**Comment by Quetzacoalt91** [comment]
Squashed & rebased

**Review by Quetzacoalt91 — CHANGES_REQUESTED** [review]
Hmm, it appears after a quick test that we cannot change the language anymore once we are on the order history. This is probably what's making the E2E test failing.

Removing QA :heavy_check_mark:

**Comment by SebBareyre** [comment]
@Quetzacoalt91 I just test to change language on an order history page and it works... what did not work for you ?

**Comment by Quetzacoalt91** [comment]
With your PR, URLs like

```
http://ip6-localhost/PrestaShop/fr/index.php?controller=order-detail&id_order=6
```

becomes

```
http://ip6-localhost/PrestaShop/index.php?controller=order-detail&id_order=6
```

We loose the reference to a language in the URL, and this prevents the customer to use another language.
To reproduce the issue, go to a order history then try to switch language. You'll see that whatever the language you choose, the URL will be the same.

capture du 2018-09-19 11-56-15

**Comment by matks** [comment]
Hi everyone !

I'm sorry I dont know why this one went under the radar for so long.

@SebBareyre In 2018 ( 😅  ) Quetzacoalt91 reported an issue in how the logic behaves, which introduces an issue. Do you feel like getting back to this PR or should I close it?

**Comment by PierreRambaud** [comment]
Hi @SebBareyre ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**